### PR TITLE
[Lib] Ensure tests run against linked packages and upgrade jest@24.8.0

### PR DESCRIPTION
**Pull request by damassi**
This fixes an issue we've observed related to running tests while yarn linking to other packages, triggering "cannot use hooks outside body of function" error. Also updates Jest to latest. 

See https://github.com/facebook/react/issues/13991 for more context.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 8efc54c730ccbb33479993e894ccb32f3756cc84

https://deploy-preview-2517--artsy-reaction.netlify.com

File: .circleci/config.yml [hunk 1]
```diff
@@ -20,7 +20,7 @@ workflows:
           requires:
             - yarn/workflow-queue
       - yarn/jest:
-          args: --runInBand
+          args: --runInBand --no-cache -u
```
Comment (damassi) [inline .circleci/config.yml]: We'll need to immediately follow up with a PR that removes these flags. Couldn't figure out how to bust circle's cache and ran into that snapshot update issue.

**Comment by alloy** [comment]
Unsure what “that snapshot issue” is, but are all these styling snapshot changes correct?

**Comment by damassi** [comment]
@alloy - sometimes when tooling changes happen snapshots on CI get out of sync with local. I'm not sure exactly why it happens or how to formally fix (since there's no longer a build without cache button on Circle) but this is how i've typically fixed it. The snapshot changes themselves are due to major version bump in Jest.

**Comment by kierangillen** [comment]
Thanks @damassi !

**Comment by artsyit** [comment]
:rocket: PR was released in v16.14.7 :rocket:

**Pull request merged**